lib/resty/radixtree.lua

```diff
@@ -307,14 +348,45 @@ function _M.new(routes)
         error("failed to new radixtree iterator")
     end
 
+    local tree_with_host = radix.radix_tree_new()
+    local tree_it_with_host = radix.radix_tree_new_it(tree_with_host)
+    if tree_it_with_host == nil then
+        error("failed to new radixtree iterator")
+    end
+
+
     local self = setmt__gc({
             tree = tree,
             tree_it = tree_it,
             match_data_index = 0,
             match_data = new_tab(#routes, 0),
             hash_path = new_tab(0, #routes),
+
+            tree_with_host = tree_with_host,
+            tree_it_with_host = tree_it_with_host,
+            match_data_index_with_host = 0,
+            match_data_with_host = new_tab(#routes, 0),
+            hash_path_with_host = new_tab(0, #routes),
         }, mt)
 
+    -- order by hosts max length, so that qqq.foo.bar.com match *.foo.bar.com not *.bar.com
+    local _get_hosts_max_item_length = function(hosts)
```

Comment:
These logics make me a bit confused, I think they are useless.

Reply:
this part of code is sort the `routes`，which want the route with longest host item at the head of array.

```
routes = {
   {
     path = '/a1/*',
     hosts = {'*.qq.com'},
   },
  {
     path = '/a2/*',
     hosts = {'*.1.qq.com'},
   }
}
```

after sort by the code below
```
    table.sort(routes, function(a,b)
          return _get_hosts_max_item_length(a.hosts) > _get_hosts_max_item_length(b.hosts)
    end)
```

the path a2 will move to the first position of array routes

Reply:
the longer host will be try to match first, so that the domain lie `1.1.qq.com` will be match `*.1.qq.com` instead of `*.qq.com`